java/src/main/native/src/TableJni.cpp

```diff
@@ -935,13 +935,70 @@ jlongArray combine_join_results(JNIEnv *env, cudf::table &left_results,
   return combine_join_results(env, std::move(left_cols), std::move(right_cols));
 }
 
+cudf::column_view remove_validity_from_col(cudf::column_view column_view) {
+  if (!cudf::is_compound(column_view.type())) {
+    if (column_view.nullable() && column_view.null_count() == 0) {
+      // null_mask is allocated but no nulls present therefore we create a new column_view without
+      // the null_mask to avoid things blowing up in reading the parquet file
+      return cudf::column_view(column_view.type(), column_view.size(), column_view.head(), nullptr,
+                               0, column_view.offset());
+    } else {
+      return cudf::column_view(column_view);
+    }
+  } else {
+    std::unique_ptr<cudf::column_view> ret;
+    std::vector<cudf::column_view> children;
+    children.reserve(column_view.num_children());
+    for (auto it = column_view.child_begin(); it != column_view.child_end(); it++) {
+      children.push_back(remove_validity_from_col(*it));
+    }
+    if (!column_view.nullable() || !column_view.null_count() == 0) {
+      ret.reset(new cudf::column_view(column_view.type(), column_view.size(), nullptr,
+                                      column_view.null_mask(), 0, column_view.offset(), children));
```

Comment:
Looks like you are missing the null count here?
```suggestion
    if (!column_view.nullable() || column_view.null_count() != 0) {
      ret.reset(new cudf::column_view(column_view.type(), column_view.size(), nullptr,
                                      column_view.null_mask(), column_view.null_count(), column_view.offset(), children));
```

or maybe
```suggestion
    if (!column_view.nullable() || column_view.null_count() != 0) {
      ret.reset(new cudf::column_view(column_view));
```

Reply:
Yes, good catch